ire/src/main/scala/hu/bme/mit/ire/stateless/unary/DuplicateEliminationNode.scala

```diff
@@ -1,14 +1,14 @@
 package hu.bme.mit.ire.stateless.unary
 
 import hu.bme.mit.ire.SingleForwarder
-import hu.bme.mit.ire.messages.{ChangeSet, ReteMessage}
+import hu.bme.mit.ire.messages.{BatchChangeSet, ReteMessage}
 import hu.bme.mit.ire.nodes.unary.UnaryNode
 
-class DuplicateEliminationNode(override val next: (ReteMessage) => Unit) extends UnaryNode with SingleForwarder {
+class DuplicateEliminationNode(override val next: (ReteMessage) => Unit) extends UnaryNode[BatchChangeSet] with SingleForwarder {
```

Comment:
Negative messages still go through this.

Reply:
What do you mean?

Reply:
If this node is in a network of ire nodes, it might receive IncerementalChangeSets with negative elements.

Reply:
Let's suppose the EngineFactory takes care of that. This would be quite cumbersome to express in the type system alone.

Reply:
Let's not. Why would we do that?

>quite cumbersome to express in the type system alone

Well it was expressed before this change.

Reply:
Any ideas? Perhaps we should introduce a trait that applies to "control messages" and "batch change sets"?

Reply:
I don't have any ideas because i still don't get what this PR is trying to achieve.